Q1. How do you comment code in Python? What are the different types of comments?

These comments are ignored by the Python interpreter and are intended for human readers to understand the code better

Q2. What are variables in Python? How do you declare and assign values to variables?

In Python, variables are used to store and manipulate data. A variable is essentially a named location in the computer's memory where you can store a value. Unlike some other programming languages, Python does not require explicit declaration of variables.

To declare and assign values to variables in Python, you simply need to choose a name for the variable and use the assignment operator (=) to assign a value to it. Here's an example:

In [5]:
# Variable declaration and assignment
x = 10
name = "John Doe"
pi = 3.14159
is_valid = True


In [7]:
print(x)
print(name)
print(pi)
print(is_valid)


10
John Doe
3.14159
True


Q3. How do you convert one data type to another in Python?

#1.int(): Converts a value to an integer data type.
x = int(3.14)  # x will be assigned the value 3

#2.float(): Converts a value to a floating-point data type.
y = float("5.7")  # y will be assigned the value 5.7

#3.str(): Converts a value to a string data type.
z = str(42)  # z will be assigned the string value "42"

#4.bool(): Converts a value to a boolean data type.
a = bool(0)  # a will be assigned the value False

#5.list(), tuple(), and set(): Convert a value to a list, tuple, or set data type, respectively.

my_list = list("hello")  # my_list will be assigned the list ['h', 'e', 'l', 'l', 'o']
my_tuple = tuple([1, 2, 3])  # my_tuple will be assigned the tuple (1, 2, 3)
my_set = set([1, 2, 3])  # my_set will be assigned the set {1, 2, 3}

 In addition to these built-in functions, some data types in Python provide their own conversion methods.
For example, the str() method can be used to convert other data types to a string:


#6.x = 42
x_str = str(x)  # x_str will be assigned the string value "42"

Similarly, the list(), tuple(), and set() constructors can be used to convert other data types to their 
respective types.

It's important to note that not all conversions are valid or meaningful.
For example, converting a string that doesn't represent a valid integer to an integer will raise a ValueError:


x = int("abc")  # Raises ValueError: invalid literal for int() with base 10: 'abc'


Q4. How do you write and execute a Python script from the command line?

1-Open a text editor and write your Python code. Save the file with a .py extension. For example, you can create a file named myscript.py and add the following code:

print("Hello, world!")

2-Open a command prompt or terminal on your computer.

3-Navigate to the directory where your Python script is located using the cd command. For example, if your script is saved in the Documents folder, you would use the following command on Windows:


cd C:\Users\YourUsername\Documents
 On macOS or Linux, you can use the following command:


cd /Users/YourUsername/Documents

4-Once you're in the correct directory, you can run the Python script using the python command followed by the name of your script. For example:

python myscript.py

5-Press the Enter key, and the Python script will be executed. In this example, it will print "Hello, world!" to the console.
6--Note: In order to run Python scripts from the command line, you need to have Python installed on your computer and added to your system's PATH environment variable. If you're using a specific Python version or virtual environment, make sure you're using the correct Python executable or activate the desired virtual environment before executing the script.

Q5. Given a list my_list = [1, 2, 3, 4, 5], write the code to slice the list and obtain the sub-list [2, 3].

In [4]:
my_list = [1, 2, 3, 4, 5]
sub_list = my_list[1:3]
print(sub_list)

[2, 3]


Q6. What is a complex number in mathematics, and how is it represented in Python?

In Python, complex numbers are represented using the complex data type. The complex type is built-in, and complex numbers can be created by directly specifying the real and imaginary parts, or by using the j or J suffix to denote the imaginary unit.

In [8]:
d=4+5j
type(d)

complex

In [10]:
d.real

4.0

In [13]:
d.imag

5.0

In [15]:
# Creating complex numbers
z1 = complex(2, 3)     # z1 = 2 + 3j
z2 = 4 + 5j            # z2 = 4 + 5j
z3 = 1.5 - 2.5j        # z3 = 1.5 - 2.5j

# Accessing real and imaginary parts
print(z1.real)         # Output: 2.0
print(z1.imag)         # Output: 3.0

# Performing arithmetic operations
z_sum = z1 + z2        # Addition: (2 + 3j) + (4 + 5j) = 6 + 8j
z_diff = z1 - z2       # Subtraction: (2 + 3j) - (4 + 5j) = -2 - 2j
z_prod = z1 * z2       # Multiplication: (2 + 3j) * (4 + 5j) = -7 + 22j
z_div = z1 / z2        # Division: (2 + 3j) / (4 + 5j) = 0.5609756097560976 + 0.0487804878048781j

# Conjugate of a complex number
z_conj = z1.conjugate()   # Conjugate of (2 + 3j) = 2 - 3j


2.0
3.0


Q7. What is the correct way to declare a variable named age and assign the value 25 to it?

In [18]:
age=25

Q8. Declare a variable named price and assign the value 9.99 to it. What data type does this variable
belong to?

In [20]:
price=9.99
print(price)
type(price)

9.99


float

Q9. Create a variable named name and assign your full name to it as a string. How would you print the
value of this variable?

In [22]:
name="sankalp prasad kushwaha"
print(name)
type(name)

sankalp prasad kushwaha


str

Q10. Given the string "Hello, World!", extract the substring "World".

In [25]:
s = "Hello, World!"
substring = s[7:12]
print(substring)

World


Q11. Create a variable named "is_student" and assign it a boolean value indicating whether you are
currently a student or not.

In [27]:
is_student = True # If you are currently a student

is_student = False  # If you are not currently a student
